### ISSUE TITLE
DEPR: remove deprecated linprog-methods

### ISSUE THREAD

**Issue by h-vetinari** [scipy/scipy#15707]
See #15512
```
def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
            bounds=None, method='interior-point', callback=None,
            bounds=None, method='highs', callback=None,
            options=None, x0=None, integrality=None):
    [...]

    Notes
    -----
    This section describes the available solvers that can be selected by the
    'method' parameter.

    `'highs-ds'` and
    `'highs-ipm'` are interfaces to the
    HiGHS simplex and interior-point method solvers [13]_, respectively.
    `'highs'` (default) chooses between
    the two automatically. These are the fastest linear
    programming solvers in SciPy, especially for large, sparse problems;
    which of these two is faster is problem-dependent.
    The other solvers (`'interior-point'`, `'revised simplex'`, and
    `'simplex'`) are legacy methods and will be removed in the second release
    after SciPy 1.9.0.
```

**Comment by EwoutH** [comment]
When reading the docs I initially read over this depreciation that's currently mentioned only in the notes, so I openend #16436 to also mention it directly in the linprog function parameters.

**Comment by j-bowhay** [comment]
@mdhaber I see you self assigned yourself to this, does this mean you wanted to do this yourself?

**Comment by mdhaber** [comment]
Yes, I'll do it. We can't execute this yet, though. Our HiGHs method needs to support callbacks before we can. 
@mckib2 did callbacks get added upstream?

**Comment by mckib2** [comment]
@mdhaber Not that I am aware of -- please correct me if I'm wrong, @jajhall.  The related upstream issues that I could find quickly are the original from us https://github.com/ERGO-Code/HiGHS/issues/911 and related to logging https://github.com/ERGO-Code/HiGHS/issues/709

If you point us in the right direction, we might be able to add the callbacks we want in our fork rather than upstreaming them

**Comment by jajhall** [comment]
No, the callbacks you want have not yet been added. I'm not confident of having the C++ skills to add them, but I'll have a go between now and mid-January, when my teaching starts

**Comment by h-vetinari** [comment]
Hey @jajhall, any update on those call-backs? We did promise removal of the old methods for the upcoming 1.11 (apparently unaware of the external dependency), so I just wanted to check. If not, it's not the end of the world, wouldn't be the first time we end up executing a deprecation later than announced.

**Comment by h-vetinari** [comment]
Gentle ping @jajhall

**Comment by jajhall** [comment]
Still busy with fixing bugs in the MIP presolve and teaching. I'll get to the call-backs this summer

**Comment by h-vetinari** [comment]
Just a gentle inquiry about how things are going, resp. at least that things don't fall off the radar (now that it's summer). :)

**Comment by HaoZeke** [comment]
> Just a gentle inquiry about how things are going, resp. at least that things don't fall off the radar (now that it's summer). :)

Callbacks have been added to Highs, they are being worked on for `highspy` (https://github.com/ERGO-Code/HiGHS/pull/1447) which in turn is to replace the existing Cython wrappers to highs (https://github.com/scipy/scipy/pull/19255). Once `highspy` is used for `linprog` and callbacks are added (soonish) then the transition to `highs` as the default can be completed and the older methods be removed for good.

**Comment by h-vetinari** [comment]
Now that the highs-update was merged (https://github.com/scipy/scipy/pull/21565), where do we stand with this whole effort? Presumably we need to integrate the callbacks here, see https://github.com/scipy/scipy/issues/19451?

**Comment by h-vetinari** [comment]
This needs https://github.com/scipy/scipy/pull/19420, which has an uncertain time horizon. In the meantime we can remove the version promise.
